api/envoy/config/core/v3/http_uri.proto

```diff
@@ -46,6 +48,9 @@ message HttpUri {
     //    cluster: jwks_cluster
     //
     string cluster = 2 [(validate.rules).string = {min_len: 1}];
+
+    // [#not-implemented-hide:]
+    udpa.core.v1.ResourceLocator cluster_resource_locator = 4;
```

Comment:
Shouldn't it be URI rather than URL if we're referring to a concrete entity like a cluster that exists in the data model, vs. describing how to fetch something?

Reply:
I don't think there's actually a distinction between a resource that already exists in the data model and a case where we will need to fetch something.  We're moving more and more to a world of on-demand resource fetching, which means that in the general case, any time we refer to a resource, we don't have any way of knowing whether we've already fetched the resource or whether referring to it will cause us to go fetch it.

I realize that today, Envoy always fetches all clusters up front.  However, that's not true of gRPC, and it will likely not be true of Envoy in the future, when we eventually add on-demand cluster fetching.  (This is where I make my periodic case for something like #8200. :) ).

To say this another way, I think that the difference between `ResourceLocator` and `ResourceName` is that the former is search criteria and the latter is a name for one specific resource.  This difference is only slightly relevant today, especially when we know that the query is for a single resource instead of a collection, but if/when we add more flexible matching for context params, I think it will become more important.